### PR TITLE
add prepareFormData to api documentation

### PR DESCRIPTION
**Pull request by bezin**
Actually just porting that stuff from the 3.2 manual to the new api documentation. I added a tiny usecase example as well.

### CODE REVIEW

**Comment by lionel-m** [comment]
Hi @bezin,
Could you use [ATX style headlines](https://github.com/contao/docs/blob/master/CONTRIBUTING.md) (e.g. # H1 or ### H3)

**Comment by bezin** [comment]
Hmm, I used the storeFormData hook as a template and as far as I see all the hooks are documented like that?

**Comment by lionel-m** [comment]
Yes but these files will be modified to follow [the general rules in the contributing file](https://github.com/contao/docs/blob/master/CONTRIBUTING.md#general-rules).

**Comment by fiedsch** [comment]
I've got the script to convert the files ready (https://gist.github.com/fiedsch/9a76e2410202ff4b30c7) but did not apply it yet. How can I avoid that the changed files would be included in my open pull request #302. Create a new branch? I don't want to mess things again :-(

**Comment by bezin** [comment]
@lionel-m allright, that's good to know. I modified my headline styles and word break according to the guidelines.

File: api/hooks/prepareFormData.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+## Parameters
+
+1. *array* `$arrSubmitted`
+
+	The user input from the form.
+
+2. *arrray* `$arrLabels`
+
+	The fields labels of the form.
```
Comment (aschempp) [inline api/hooks/prepareFormData.md]: I think _fields_ should be singular. "Field labels".

File: api/hooks/prepareFormData.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+3. *Form* `$objForm`
+
+	The form module instance.
```
Comment (aschempp) [inline api/hooks/prepareFormData.md]: Please change this to "The form instance.". The `Form` class is actually a _hybrid_, it's not a frontend module…

File: api/hooks/prepareFormData.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+// config.php
+$GLOBALS['TL_HOOKS']['prepareFormData'][] = array('MyClass', 'myPrepareFormData');
+
+// MyClass.php
+public function myPrepareFormData(&$arrSubmitted, $arrLabels, $objForm)
+{
+    // this calculates a deadline from a given timestamp
+    // and stores it as deadline in $arrSubmitted 
+    $arrSubmitted['deadline'] = $strtotime('+1 hour', $arrSubmitted('tstamp'));
```
Comment (aschempp) [inline api/hooks/prepareFormData.md]: The $ in front of `strtotime` seems wrong to me?

Comment (aschempp) [inline api/hooks/prepareFormData.md]: Also, it should be `$arrSubmitted['tstamp']` (square brackets)

File: api/hooks/prepareFormData.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+}
+```
+
+
+## References
```
Comment (aschempp) [inline api/hooks/prepareFormData.md]: Please add a _More Information_ level 2 headline before this, and change _References_ to a level 3 (same as _See Also_). See https://github.com/contao/docs/pull/292#issuecomment-110245673 for the discussion about that.

File: api/hooks/prepareFormData.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+# prepareFormData
+
+The `prepareFormData` hook is triggered after a form has been submitted, but 
+before it is processed. It passes the form data array, the form labels array 
+and the form object as arguments and does not expect a return value. This way 
+the data can be changed or extended, prior to execution of actions like email 
+distribution or data storage. It is available from version 3.0.0.
+
```
Comment (aschempp) [inline api/hooks/prepareFormData.md]: Thanks to GitBook, the version information can now be shown better. Please change it to the following on a new line:

`> **Tag** Available from version 3.0.0.`

This wil create a nice balloon in the HTML output.

**Comment by aschempp** [comment]
@bezin thanks for the work! Please also update the `hooks.md` document to link to this new documentation.

@fiedsch do you mean merging this would affect your gist script? Or because of another rebase/merge?

**Comment by aschempp** [comment]
> Please also update the hooks.md document to link to this new documentation.

Stupid me, I just noticed you already did that :D

**Comment by fiedsch** [comment]
@aschempp I'd apply the script to the files once you merged @bezin's PR after updating (rebasing) my repo. If the style is already ATX the script should simply do nothing. But I'll double check before creating any PR. One thing I have not found out yet: how do I separate this from my current PR? If I change files in my repo the changes become part of the current open PR. Do I have to create a new branch?

**Comment by aschempp** [comment]
yes you should alway create a new branch for anything you're working on. You don't have to wait for the PR to be merged, it's GITs job to merge both. Shouldn't be a problem because your PR won't affect the other file.

**Comment by bezin** [comment]
I needed to restructure my repo a little bit, sorry for the hassle. I open a new PR.

**Pull request closed without merging**